staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -2916,6 +2916,31 @@ const (
 	ServiceAffinityNone ServiceAffinity = "None"
 )
 
+const (
+	// DefaultClientIPServiceAffinitySeconds is the default timeout seconds
+	// of Client IP based session affinity - 3 hours.
+	DefaultClientIPServiceAffinitySeconds int32 = 10800
+	// DefaultClientIPServiceAffinitySeconds is the max timeout seconds
+	// of Client IP based session affinity - 1 day.
+	MaxClientIPServiceAffinitySeconds int32 = 86400
+)
+
+// SessionAffinityConfig represents the configurations of session affinity.
+type SessionAffinityConfig struct {
+	// ClientIP contains the configurations of Client IP based session affinity.
```

Comment:
In godoc for external apis, refer to the field by its JSON name (`clientIP`) because this is documentation that is output into our formal human readable documentation, where the Go-field name doesn't make sense.